plugins/woocommerce/tests/php/includes/rest-api/Controllers/Version4/Products/class-wc-rest-products-v4-controller-tests.php

```diff
@@ -0,0 +1,1994 @@
+<?php
+
+use Automattic\WooCommerce\Enums\ProductStatus;
+use Automattic\WooCommerce\Enums\ProductType;
+use Automattic\WooCommerce\Internal\CostOfGoodsSold\CogsAwareUnitTestSuiteTrait;
+
+/**
+ * class WC_REST_Products_Controller_Tests.
+ * Product Controller tests for V4 REST API.
+ */
+class WC_REST_Products_V4_Controller_Test extends WC_REST_Unit_Test_Case {
+	use CogsAwareUnitTestSuiteTrait;
+
+	/**
+	 * Enable the experimental REST API feature.
+	 */
+	private function enable_experimental_rest_api_feature() {
+		add_filter(
+			'woocommerce_admin_features',
+			function ( $features ) {
+				$features[] = 'experimental-wc-rest-api';
+				return $features;
+			}
+		);
+	}
+
+	/**
+	 * Disable the experimental REST API feature.
+	 */
+	private function disable_experimental_rest_api_feature() {
+		add_filter(
+			'woocommerce_admin_features',
+			function ( $features ) {
+				$features = array_diff( $features, array( 'experimental-wc-rest-api' ) );
+				return $features;
+			}
+		);
+	}
+
+	/**
+	 * Runs after each test.
+	 */
+	public function tearDown(): void {
+		parent::tearDown();
+		$this->disable_cogs_feature();
+		$this->disable_rest_api_v4_feature();
+	}
+
+	/**
+	 * Enable the REST API v4 feature.
+	 */
+	public static function enable_rest_api_v4_feature() {
+		add_filter(
+			'woocommerce_admin_features',
+			function ( $features ) {
+				$features[] = 'rest-api-v4';
+				return $features;
+			},
+		);
+	}
+
+	/**
+	 * Disable the REST API v4 feature.
+	 */
+	public static function disable_rest_api_v4_feature() {
+		add_filter(
+			'woocommerce_admin_features',
+			function ( $features ) {
+				$features = array_diff( $features, array( 'rest-api-v4' ) );
+				return $features;
+			}
+		);
+	}
```

Comment:
Copied pasted from `plugins/woocommerce/tests/php/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller-tests.php`. I added only `enable_rest_api_v4_feature` and `disable_rest_api_v4_feature` to ensure that the feature flag is enabled before running unit tests.